operations/observability/Observability.md

```diff
@@ -0,0 +1,300 @@
+The service for the grok-exporter is created for access at port `9144`. Service-monitor is created for Prometheus to access the metrics that are created by the grok-exporter.
+
+### Configuration file for the grok-exporter
+
+Created the configuration file for the grok-exporter to parse the Goss test log file and make metrics from them.
+We are using grok-exporter version 3 configuration for this task and matching log expressions with regular expression.
+
+The following is the example configuration passed to the grok-exporter to get metrics:
+
+```bash
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example_name
+  namespace: example_namespace
+data:
+  config.yml: |-
+    global:
+      config_version: 3
+    input:
+      type: file
+      paths: 
+      - /logs/goss_tests/*
+      fail_on_missing_logfile: false
+    grok_patterns:
+      - 'REGEX [regular].*expression'
+      - 'EXAMPLE [aA-zZ-].*'
+    metrics:
+      - type: gauge
+        name: example_metric
+        help: EXAMPLE METRIC HELP
+        match: '{"log_file": "%{REGEX:logfile}", "TEST": "%{EXAMPLE:ex}"}'
+        value: '{{`{{.logfile}}`}}'
+        labels:
+          example_test: '{{`{{.ex}}`}}'
+    server:
+     port: 9144
+```
+
+### Prometheus metrics and Grafana dashboard
+
+After the preceding steps the `goss_tests` metrics are seen in Prometheus when Goss tests are run. Using these metrics, Grafana dashboards are created to shows the Goss tests details visually.
+
+Goss test dashboard features:
+
+- The dashboard as dropdown for product, suite, and tests.
+- By default, all products, suites and tests are selected for overall Goss tests result.
+- The overall product result, total number of the products, products passed, products failed, and its execution time is seen.
+- Suite results and test results are seen from the dashboard.
+- Node-wise test result are seen with failed nodes, passed nodes, and the suite/test description.
+
+![Goss test dashboard](../../img/operations/GossTestsDashboard.png "Goss Test Dashboard")
+
+## Error dashboard
+
+Error dashboards provide key glance insights as to where and what is broken and needs attention. This works for both internal HPE environments and clusters as well as is a containerized shippable solution that customers can use as well.
+
+The automatic generation of the desired Prometheus grok-exporter configuration based upon passing in a set of regular expression patterns to detect.
+This includes but is not limited to regular expression patterns to match and generate metrics on phases of manual administrator steps, as well as phases of installation scripts and automation.
+
+### Error dashboards features
+
+- Automate Grafana dashboards with error and failure message during CSM upgrade and install.
+- Create dashboard for issues comparison across multiple dimensions and clusters.
+- Monitor ConMan logs from `csm-pit`.  
```

Comment:
Does this mean the PIT node?

Reply:
yes

Reply:
It should use that language then. `csm-pit` indicates a thing literally named `csm-pit`.
```suggestion
- Monitor ConMan logs from the PIT node.
```

Reply:
Changed csm-pit to PIT node as asked.